idx/memory/memory.go

```diff
@@ -87,6 +89,23 @@ func ConfigProcess() {
 	}
 }
 
+// interface implemented by both MemoryIdx and PartitionedMemoryIdx
+// this is needed to support unit tests.
+type MemoryIndex interface {
```

Comment:
I think it seems a bit confusing that there's now a thing called `MemoryIndex` and another thing called `MemoryIdx`. Wouldn't it make sense to rename `MemoryIdx` into something like `UnpartitionedMemoryIndex` or `MonolithicMemoryIndex` or whatever else describes the different between it and the `PartitionedMemoryIndex`?